### PR TITLE
Implement specific CORS policy to limits allowed origins for the API

### LINKED ISSUE #30: [BUG] Lock down CORS
**Describe the bug**

The API is currently open to abuse and the CORS need to be locked down. Exploitation will bring the service down however we're not on consumption so the risk here is managed. 

**To Reproduce**
Check the API. CORS policy is open for all origins and all verbs

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Windows, macOS]

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by davidsmonkeys**
## Description

Limits the Allowed Origins for the API

Implement specific CORS policy from appsettings.json
- Introduce new CORS policy configuration in `Startup.cs` to read allowed origins from `appsettings.json`.
- Add `Microsoft.IdentityModel.Tokens` namespace in `Startup.cs`.
- Replace `app.UseCors("AllowAll")` with `app.UseCors("AllowSpecificOrigins")`.
- Update `appsettings.json` with a new `Cors` section and `AllowedOrigins` key.

**IMPORTANT** - With the addition to the appconfiguration.json file we need to add these settings into the AzureAppConfiguration settings for test and Live with the correct values. 

(Label: Staging) Cors:AllowedOrigins - https://staging.dertinfo.co.uk, https://staging-app.dertinfo.co.uk
(Label: Production) Cors:AllowedOrigins - https://www.dertinfo.co.uk, https://app.dertinfo.co.uk

Fixes #30  (issue)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

I have tested locally and validated that the CORS policy allows the web origin when it should and when running on a different port access is denied. 

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules 
### :bulb: PR Summary generated by FirstMate

**CORS Policy Implementation:**
- Implemented a specific CORS policy to limit allowed origins for the API.

**Changes:**
*Startup Configuration:*
- Replaced "AllowAll" CORS policy with "AllowSpecificOrigins" to restrict origins.
- Added validation for allowed origins configuration in `Startup.cs`.
- Throws exceptions for empty or invalid URLs in allowed origins.

*Configuration Update:*
- Added "Cors" section in `appsettings.json` with allowed origins for local development.

**TLDR**: This PR implements a specific CORS policy to restrict allowed origins for the API, enhancing security. Review the changes in `Startup.cs` for policy implementation and `appsettings.json` for configuration details.

<sub><sup>Generated by FirstMate and automatically updated on every commit.</sup></sub>

### CODE REVIEW

File: src/dertinfo-api/Startup.cs [hunk 1]
```diff
@@ -142,7 +154,7 @@ public void Configure(IApplicationBuilder app, DertInfoContext context)
 
             // Runs matching. An endpoint is selected and set on the HttpContext if a match is found.
             app.UseRouting();
-            app.UseCors("AllowAll");
+            app.UseCors("AllowSpecificOrigins");
 
             // Middleware that run after routing occurs. Usually the following appear here:
```
[the remark this answers is not in the document]
Comment (davidsmonkeys) [reply src/dertinfo-api/Startup.cs:159]: Added a page in the wiki https://github.com/dertinfo/dertinfo/wiki/Security to outline what we're doing with the CORS policy

**Review by dertinfo — COMMENTED** [review]
Great support from the first mate bot.

**Review by dertinfo — APPROVED** [review]
Approving changes

**Comment by davidsmonkeys** [comment]
**Note** - When deploying this change we had problems in staging where the extension version of the .NET runtime were 8.0.10 and the site was showing 500.13. I needed to upgrade the extensions to 8.0.11 for the site to work. 

It'm unsure why I'm having this problem. It's happened twice now where I've had to upgrade the runtime. Or what the change has been to cause this. It appears that it's happening on deployment since the .NET 8 upgrade. Perhaps the way the API is packaged has changed. Requires Review
